### PR TITLE
added new unittests for alert and alerts

### PR DESCRIPTION
**Pull request by pbubas**
adding new pytests

### CODE REVIEW

File: tests/__init__.py [hunk 1]
```diff
@@ -5,3 +5,121 @@
...
+               "testName": "BGP test name",
+               "violationCount": 61,
+               "type": "BGP",
+               "severity": "MINOR"
+          }
+]
+}
+
+ALERT_LIST = {}
+ALERT_LIST["alert"]=[ALERT_WITH_AGENTS["alert"][0],ALERT_WITH_MONITORS["alert"][0]]
```
Comment (Drrechu) [inline tests/__init__.py]: Why creating a dict with on key and call variable a list. Just create a list
```suggestion
ALERT_LIST = [ALERT_WITH_AGENTS["alert"][0], ALERT_WITH_MONITORS["alert"][0]]
```

File: tests/__init__.py [hunk 2]
```diff
@@ -5,3 +5,121 @@
 USERNAME = 'foouser@foo.com'
 AUTH_TOKEN = 'fooauthtoken'
 BEARER_TOKEN = 'bearer token'
+ALERT_WITH_AGENTS={'alert': [ {
```
Comment (Drrechu) [inline tests/__init__.py]: Fix the styling to match something like this
```python
ALERT_WITH_AGENTS = {
    'alert': [
        {
            "active": 0,
            "agents": [
                {
                    "active": 0,

                },
                {
                    "active": 0,
                }
            ],
            "alertId": 111111111,
            "apiLinks": [
                {
                    "rel": "related",
                    "href": "https://api.thousandeyes.com/v6/tests/4443322"
                },
                {
                    "rel": "data",
                    "href": "https://api.thousandeyes.com/v6/web/http-server/4443322"
                }
            ],
            "permalink": "https://app.thousandeyes.com/alerts/list/?__a=333666&alertId=111111111",
            "severity": "INFO"
        }
    ]
}
```

File: tests/__init__.py [hunk 3]
```diff
@@ -5,3 +5,121 @@
...
+        "testId": 4443322,
+        "testName": "HTTP test to server",
+        "violationCount": 2,
+        "type": "HTTP Server",
+        "severity": "INFO"
+    }
+    ]
+    }
+
+ALERT_WITH_MONITORS={"alert": [
```
Comment (Drrechu) [inline tests/__init__.py]: ditto

File: tests/test_alert.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+from unittest import TestCase
+from unittest.mock import patch
+from thousandeyessdk import ThousandEyes as TE
+from thousandeyessdk.enum import AlertType
+
+from . import ALERT_WITH_AGENTS, ALERT_WITH_MONITORS, USERNAME, AUTH_TOKEN, URL, USERNAME
+
+@patch('thousandeyessdk.requests.request')
+class TestAlert(TestCase):
+    def test_alert_with_agents(self, response_mock):
```
Comment (Drrechu) [inline tests/test_alert.py]: response_mock is an request object mock so name it accordingly, I suggest m__request

File: tests/test_alert.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        response_mock().json.return_value = ALERT_WITH_AGENTS
+        response_mock().ok=True
+        te_client = TE(username=USERNAME, auth_token=AUTH_TOKEN)
```
Comment (Drrechu) [inline tests/test_alert.py:14]: Please fallow test styling:

preparation, empty line, tested method call, empty line, assertions

File: tests/test_alerts.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+
+class TestAlerts(TestCase):
+
+    @patch('thousandeyessdk.requests.request')
+    def test_alerts_ids(self, response_mock):
+        response_mock().json.return_value = ALERT_LIST
+        response_mock().ok = True
+        te_client = TE(username=USERNAME, auth_token=AUTH_TOKEN)
+        alerts = alerts = te_client.alerts.list()
```
Comment (Drrechu) [inline tests/test_alerts.py]: Why not assigning it three times? :D

File: tests/test_alert.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        alert = te_client.alerts.get(ALERT_WITH_AGENTS['alert'][0]['alertId'])
+        assert alert.id == 111111111
+        assert alert.active == False
```
Comment (Drrechu) [inline tests/test_alert.py]:
```suggestion
        assert alert.active is False
```

File: tests/test_alert.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        assert alert.inactive == True
```
Comment (Drrechu) [inline tests/test_alert.py]:
```suggestion
        assert alert.inactive is True
```

File: tests/test_alert.py [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+        assert alert.disabled == False
+        assert alert.rule_expression == "((responseTime >= 300 ms))"
+        assert alert.type.value == "HTTP Server"
+        assert alert.type.name == "HTTP_SERVER"
+        assert alert.type == AlertType("HTTP Server")
+        assert alert.string_type == "HTTP Server"
+        assert alert.date_start == None # date_start not supported currently
```
Comment (Drrechu) [inline tests/test_alert.py]:
```suggestion
        assert alert.date_start is None
```

**Review by Drrechu — CHANGES_REQUESTED** [review]

File: tests/__init__.py (hunk 1 above)
Comment (pbubas) [reply tests/__init__.py]: suggested change does not work.
Alert list must be defined in "alert" key

File: tests/__init__.py (hunk 2 above)
Comment (pbubas) [reply tests/__init__.py]: fixed b8a3076

File: tests/__init__.py (hunk 3 above)
Comment (pbubas) [reply tests/__init__.py]: fixed b8a3076

File: tests/test_alert.py (hunk 4 above)
Comment (pbubas) [reply tests/test_alert.py]: corrected a6c07d7

File: tests/test_alerts.py (hunk 6 above)
Comment (pbubas) [reply tests/test_alerts.py]: corrected d0d8ff2

File: tests/test_alert.py (hunk 7 above)
Comment (pbubas) [reply tests/test_alert.py]: corrected fb3d2b2

File: tests/test_alert.py (hunk 5 above)
Comment (pbubas) [reply tests/test_alert.py:14]: corrected

File: tests/__init__.py (hunk 1 above)
Comment (Drrechu) [reply tests/__init__.py]: Ok then 
```suggestion
ALERT_LIST={"alert": [ALERT_WITH_AGENTS["alert"][0],ALERT_WITH_MONITORS["alert"][0]]}
```

File: tests/test_alert.py [hunk 10]
```diff
@@ -0,0 +1,44 @@
+from unittest import TestCase
+from unittest.mock import patch
+from thousandeyessdk import ThousandEyes as TE
+from thousandeyessdk.enum import AlertType
+
+from . import ALERT_WITH_AGENTS, ALERT_WITH_MONITORS, USERNAME, AUTH_TOKEN, URL, USERNAME
```
Comment (Drrechu) [inline tests/test_alert.py]: Some unused imports here

File: tests/test_alert.py [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
```
Comment (Drrechu) [inline tests/test_alert.py:7]: Add empty line

**Review by Drrechu — CHANGES_REQUESTED** [review]

File: tests/__init__.py (hunk 1 above)
Comment (pbubas) [reply tests/__init__.py]: ok

File: tests/test_alert.py (hunk 9 above)
Comment (pbubas) [reply tests/test_alert.py]: removed commetns

File: tests/test_alert.py (hunk 11 above)
Comment (pbubas) [reply tests/test_alert.py:7]: ok removed URL as not used

**Review by akinard — APPROVED** [review]
Approved.

**Review by Drrechu — APPROVED** [review]

**Pull request merged**